src/superpmi-collect/superpmicollect.cs

```diff
@@ -0,0 +1,777 @@
+        private static string s_finalFailMclFile = null;    // Pathname for a temporary .MCL file used for noticing superpmi replay failures against final MCH.
+        private static string s_baseMchFile = null;         // The base .MCH file path
+        private static string s_cleanMchFile = null;        // The clean .MCH file path
+        private static string s_finalMchFile = null;        // The clean thin unique .MCH file path
+        private static string s_tocFile = null;             // The .TOC file path for the clean thin unique .MCH file
+        private static string s_errors = "";                // Collect non-fatal file delete errors to display at the end of the collection process.
+
+        private static bool s_saveFinalMchFile = false;     // Should we save the final MCH file, or delete it?
+
+        private static void SafeFileDelete(string filePath)
+        {
+            try
+            {
+                File.Delete(filePath);
+            }
+            catch(Exception ex)
+            {
+                string err = string.Format("Error deleting file \"{0}\": {1}", filePath, ex.Message);
+                s_errors += err + System.Environment.NewLine;
+                Console.Error.WriteLine(err);
+            }
+        }
+
+        private static void CreateTempDirectory(string tempPath)
+        {
+            if (tempPath == null)
+            {
+                tempPath = Path.GetTempPath();
+            }
+            s_tempDir = Path.Combine(tempPath, Path.GetRandomFileName() + "SPMI");
+            if (Directory.Exists(s_tempDir))
+            {
+                throw new SpmiException("temporary directory already exists: " + s_tempDir);
+            }
+            DirectoryInfo di = Directory.CreateDirectory(s_tempDir);
+        }
+
+        private static void ChooseFilePaths(string outputMchPath)
+        {
+            s_baseFailMclFile  = Path.Combine(s_tempDir, "basefail.mcl");
+            s_finalFailMclFile = Path.Combine(s_tempDir, "finalfail.mcl");
+            s_baseMchFile      = Path.Combine(s_tempDir, "base.mch");
+            s_cleanMchFile     = Path.Combine(s_tempDir, "clean.mch");
+
+            if (outputMchPath == null)
+            {
+                s_saveFinalMchFile = false;
+                s_finalMchFile = Path.Combine(s_tempDir, "final.mch");
+                s_tocFile = Path.Combine(s_tempDir, "final.mch.mct");
+            }
+            else
+            {
+                s_saveFinalMchFile = true;
+                s_finalMchFile = Path.GetFullPath(outputMchPath);
+                s_tocFile = s_finalMchFile + ".mct";
+            }
+        }
+
+        private static int RunProgram(string program, string arguments)
+        {
+            if (program.LastIndexOf(".py") != -1)
+            {
+                string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                arguments = "-u " + program; 
```

Comment:
Do we completely drop arguments then? I don't really like the overwritting of arguments